core/src/blocktree.rs

```diff
@@ -193,66 +193,69 @@ impl Blocktree {
         false
     }
 
-    // silently deletes all blocktree column families starting at the given slot
-    fn delete_all_columns(&self, starting_slot: u64) {
-        match self.meta_cf.force_delete_all(Some(starting_slot)) {
+    /// Silently deletes all blocktree column families starting at the given slot until the `to` slot
+    /// Use with care; does not check for integrity and does not update slot metas that
+    /// refer to deleted slots
+    pub fn delete_all_columns(&self, from_slot: u64, to_slot: Option<u64>) {
+        let to_index = to_slot.map(|slot| (slot + 1, 0));
+        match self.meta_cf.force_delete(Some(from_slot), to_slot) {
             Ok(_) => (),
             Err(e) => error!(
                 "Error: {:?} while deleting meta_cf for slot {:?}",
-                e, starting_slot
+                e, from_slot
```

Comment:
this form might collapse some with map_err()

Reply:
I think clippy will complain if I map_err and just print, and map it to itself.

Reply:
I could `if let Err(e)`...